supabase/migrations/20230111145014_initial_schema.sql

```diff
@@ -6,6 +6,7 @@ CREATE TYPE issue_status AS ENUM (
 );
 
 CREATE TABLE users (
+    id SERIAL PRIMARY KEY,
```

Comment:
I wanted to use their GitHub username as the primary key. I'm no database expert but when I was working through this same problem, this greatly simplified things. 

To clarify, our entire system is dependent on GitHub with no plans to move to another platform. With this in mind, you will probably save a lot of pain and headache down the road using the GitHub ID as the primary key representing users in our system.

Reply:
One unrelated comment I also want to add concerning the User schema, in my scraper I prefixed properties with _ because the data from the UI could have collisions with the user data from their API. But there are also situations where the data is the same but represented differently. 

Off hand I remember follower and following counts from the UI are suffixed in K notation (i.e. `12.9k followers` in the UI vs `12965 followers` from the API.)

The benefit of storing the data from the UI instead of going to the API always is rate limiting. There is none for UI scrapes so it was faster and easier for me to scrape thousands of contacts. I thought it would be easiest to "enrich" the UI scraped data with....your favorite...a CRON job that can ping the API slowly over time and enrich everything.

Reply:
@0xcodercrane

Reply:
> I wanted to use their GitHub username as the primary key. I'm no database expert but when I was working through this same problem, this greatly simplified things.
> 
> To clarify, our entire system is dependent on GitHub with no plans to move to another platform. With this in mind, you will probably save a lot of pain and headache down the road using the GitHub ID as the primary key representing users in our system.

Agree with your experience.  In general, I've been using SERIAL type to make indexing faster. But I guess we will not have much difference in terms of search speed.  so I will make username primary key

Reply:
> One unrelated comment I also want to add concerning the User schema, in my scraper I prefixed properties with _ because the data from the UI could have collisions with the user data from their API. But there are also situations where the data is the same but represented differently.
> 
> Off hand I remember follower and following counts from the UI are suffixed in K notation (i.e. `12.9k followers` in the UI vs `12965 followers` from the API.)
> 
> The benefit of storing the data from the UI instead of going to the API always is rate limiting. There is none for UI scrapes so it was faster and easier for me to scrape thousands of contacts. I thought it would be easiest to "enrich" the UI scraped data with....your favorite...a CRON job that can ping the API slowly over time and enrich the contact data.

I guess we can prefix properties with _ but both represents the same number so in case UI scrapper, if we add a simple conversion method, we can get 12900 easily. Also I guess we will not have rate limiting for bounty bot because its works based on event which will have most fields we need. 

In worst case, if we couldn't use API, we can update table by creating a new migration.

Reply:
> we can get 12900 easily

Yes I already prototyped this in a Google Sheet when running some analytics on the data but I want precision so this strategy is not acceptable.

The GitHub API limits, if I recall correctly, is 60 times per minute; I agree that rate limiting is unlikely to be an issue.